src/main/java/gregification/tfc/TFCModule.java

```diff
@@ -0,0 +1,76 @@
+package gregification.tfc;
+
+import gregification.base.ModIDs;
+import gregification.base.Module;
+import gregification.tfc.food.ITFCFoodComponent;
+import gregification.tfc.food.TFCFoodComponent;
+import gregtech.api.items.metaitem.FoodStats;
+import gregtech.api.items.metaitem.FoodUseManager;
+import gregtech.api.items.metaitem.MetaItem;
+import gregtech.api.items.metaitem.stats.IFoodBehavior;
+import gregtech.api.items.metaitem.stats.IItemComponent;
+import gregtech.api.unification.OreDictUnifier;
+import gregtech.api.unification.material.Materials;
+import gregtech.api.unification.ore.OrePrefix;
+import gregtech.api.util.RandomPotionEffect;
+import gregtech.common.items.MetaItems;
+import net.dries007.tfc.api.capability.food.CapabilityFood;
+import net.dries007.tfc.api.capability.food.FoodData;
+import net.dries007.tfc.api.capability.food.FoodHandler;
+import net.minecraft.init.Items;
+import net.minecraft.init.MobEffects;
+import net.minecraft.item.EnumAction;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+import net.minecraftforge.event.AttachCapabilitiesEvent;
+import net.minecraftforge.event.RegistryEvent;
+import net.minecraftforge.fml.common.Loader;
+import net.minecraftforge.fml.common.event.FMLInitializationEvent;
+import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Collections;
+import java.util.List;
+
+@Module.Root(name = "Gregification: Ex TerraFirmaCraft")
+public class TFCModule implements Module {
+    @Module.Log
+    public static Logger logger;
+
+    @Override
+    public boolean isModuleActive() {
+        return Loader.isModLoaded(ModIDs.MODID_TFC);
```

Comment:
There should be a top-level config in `BaseConfig` that enables/disables the entire module, and is checked here